develop/tutorials/articles/192-item-selector/01-selecting-entities-using-the-item-selector-api.markdown

```diff
@@ -0,0 +1,356 @@
+    code:
+
+        import com.liferay.item.selector.ItemSelector;
+    
+        @Reference(unbind = "-")
+    
+        public void setItemSelector(ItemSelector itemSelector) {
+    
+            _itemSelector = itemSelector;
+    
+        }
+    
+        private ItemSelector _itemSelector;
+
+2.  Retrieve the URL for the Item Selector next.
+
+The URL to the item selector is obtained by invoking the method 
+`getItemSelectorURL` with the following parameters:
+
+- `RequestBackedPortletURLFactory`: factory to create portlet URLs. It
+    can be created by invoking with either `HttpServletRequest` or
+    `PortletRequest`. Use the following pattern to create your URL:
+    `RequestBackedPortletURLFactoryUtil.create(request)`.
+- `ItemSelectedEventName`: a unique JavaScript event name that is
+    triggered by the item selector when the element is selected.
+- `ItemSelectorCriterion`: the criterion (or array of criteria) that
+    specifies the element that should be selected.
+    
+Follow the pattern below to obtain the URL for the Item Selector:
+
+    List<ItemSelectorReturnType> desiredItemSelectorReturnTypes =  new
+    ArrayList<>();
+    
+    desiredItemSelectorReturnTypes.add(new *ItemSelectorReturnType());
+    
+    *ItemSelectorCriterion class *nameItemSelectorCriterion =   new
+    *ItemSelectorCriterion();
+    
+    *nameItemSelectorCriterion.setDesiredItemSelectorReturnTypes(
+    desiredItemSelectorReturnTypes);
+    
+    PortletURL itemSelectorURL = `_itemSelector.getItemSelectorURL`(
+    
+      requestBackedPortletURLFactory, "ItemSelectedEventName",
+    
+      *ItemSelectorCriterion);
+
+For example, the following configuration obtains the item selector URL for the
+`ImageItemSelectorCriterion`:
+
+    List<ItemSelectorReturnType> desiredItemSelectorReturnTypes =  new
+    ArrayList<>();
+    
+    desiredItemSelectorReturnTypes.add(new URLItemSelectorReturnType());
+    
+    ImageItemSelectorCriterion imageItemSelectorCriterion =   new
+    ImageItemSelectorCriterion();
+    
+    imageItemSelectorCriterion.setDesiredItemSelectorReturnTypes(
+    desiredItemSelectorReturnTypes);
+    
+    PortletURL itemSelectorURL = `_itemSelector.getItemSelectorURL`(
+    
+      requestBackedPortletURLFactory, "sampleTestSelectItem",
+    
+      imageItemSelectorCriterion);
+
+The return types are declared in an `ArrayList`. The `URLItemSelectorReturnType`
+is added to the list of desired return types, and the 
+`imageItemSelectorCriterion` is set as the kind of element to select. The 
+`ImageItemSelectorCriterion()` constructor is referenced so it can be used. The 
+return types are then set to the item selector criterion using the setter 
+`setDesiredItemSelectorReturnTypes()`. Finally, the item selector URL is created
+using the `getItemSelectorURL()` method, with the value of the JavaScript item 
+selected event name set to `sampleTestSelectItem`.
```

Comment:
Most of our cases are invoking the getItemSelectorURL in ConfigContributor.classes because we are using it to open a dialog selector from the editors.

However, in Journal we are invoking getItemSelectorURL from the markup. You can see it here https://github.com/liferay/liferay-portal/blob/master/modules/apps/web-experience/journal/journal-web/src/main/resources/META-INF/resources/article/display_page.jsp#L123